nrt/data/__init__.py

```diff
@@ -86,18 +87,274 @@ def mre_crit_table():
-        ...                         .get(str(period))\
-        ...                         .get(functional)
+                                    .get(str(period))\
+                                    .get(functional)
         >>> sig_level = crit_table.get('sig_levels')
         >>> crit_level = np.interp(1-alpha, sig_level, crit_values)
     """
     with open(os.path.join(data_dir, "mreCritValTable.json")) as crit:
         crit_table = json.load(crit)
     return crit_table
+
+
+def make_ts(dates, break_idx=-1, intercept=0.7, amplitude=0.15, magnitude=0.25,
+            recovery_time=1095, sigma_noise=0.02, n_outlier=3,
+            outlier_value=-0.1, n_nan=3):
+    """Simulate a harmonic time-series with optional breakpoint, noise and outliers
+
+    The time-series is generated by adding;
+    - an intercept/trend component which varies depending on the phase of the time-series
+    (stable, recovery)
+    - An annual seasonal component
+    - Random noise drawn from a normal distribution (white noise)
+    Optional outliers are then added to randomly chosen observation as well as ``np.Nan`` values.
+    Note that the seasonal cycles simulation approach used here is rather simplistic,
+    using a sinusoidal model and therefore assuming symetrical and regular behaviour
+    around the peak of the simulated variable. Actual vegetation signal is often more
+    asymetrical and irregular.
+
+    Args:
+        dates (array-like): List or array of dates (numpy.datetime64)
+        break_idx (int): Breakpoint index in the date array provided. Defaults to
+            ``-1``, corresponding to a stable time-series
+        intercept (float): Intercept of the time-series
+        amplitude (float): Amplitude of the harmonic model (note that at every point
+            of the time-series, the actual model amplitude is multiplied by the intercept
+        magnitude (float): Break magnitude (always a drop in y value)
+        recovery_time (int): Time (in days) to recover the initial intersect value
+            following a break
+        sigma_noise (float): Sigma value of the normal distribution (mean = 0) from which
+            noise values are drawn
+        n_outlier (int): Number of outliers randomly assigned to observations of the
+            time-series
+        outlier_value (float): Value to assign to outliers
+        n_nan (int): Number of ``np.nan`` (no data) assigned to observations of the
+            time-series
+
+    Example:
+        >>> from nrt import data
+        >>> import numpy as np
+        >>> import matplotlib.pyplot as plt
+
+        >>> dates = np.arange('2018-01-01', '2022-06-15', dtype='datetime64[W]')
+        >>> ts = data.make_ts(dates=dates, break_idx=30)
+
+        >>> plt.plot(dates, ts)
+        >>> plt.show()
+
+    Returns:
+        np.ndarray: Array of simulated values of same size as ``dates``
+    """
+    timestamps = (dates - np.datetime64(0, 'D')).astype(int)
```

Comment:
The code right now works only with `dtype='datetime64[D]'`, more precise dt formats will not give you days when converting to int, so it might make sense to first convert the dates array to `dtype='datetime64[D]'`
```suggestion
    dates = dates.astype('datetime64[D]')
    timestamps = dates.astype(int)
```